Makefile

```diff
@@ -0,0 +1,20 @@
+.PHONY: all
+all:
+
+.PHONY: tidy
+tidy:
+	black ./
+
+.PHONY: flake8
+flake8:
+	flake8 $$(git ls-files "*.py")
+
+.PHONY: checkstyle
+checkstyle: flake8
+
+.PHONY: test-online
+test-online:
+	python3 ./downloader.py
```

Comment:
why do you consider this a "test"? Most likely it would just fail because we cannot define sane defaults for a proper functional test. I think this is misleading and should not be included.

Reply:
Sure. As A stated in the subject, feel free to cherry pick. But I can also create pull requests one by one for the parts that you do like

Reply:
excluded this for now from this PR